x-pack/platform/plugins/shared/lens/public/react_embeddable/helper.ts

```diff
@@ -51,25 +51,52 @@ export function createEmptyLensState(
 // Make sure to inject references from the container down to the runtime state
 // this ensure migrations/copy to spaces works correctly
 export async function deserializeState(
-  attributeService: LensAttributesService,
+  {
+    attributeService,
+    ...services
+  }: Pick<
+    LensEmbeddableStartServices,
+    | 'attributeService'
+    | 'data'
+    | 'dataViews'
+    | 'data'
+    | 'visualizationMap'
+    | 'datasourceMap'
+    | 'uiSettings'
+  >,
   rawState: LensSerializedState,
   references?: SavedObjectReference[]
 ) {
+  const fallbackAttributes = createEmptyLensState().attributes;
   if (rawState.savedObjectId) {
     try {
       const { attributes, managed, sharingSavedObjectProps } =
         await attributeService.loadFromLibrary(rawState.savedObjectId);
       return { ...rawState, attributes, managed, sharingSavedObjectProps };
     } catch (e) {
       // return an empty Lens document if no saved object is found
-      return { ...rawState, attributes: createEmptyLensState().attributes };
+      return { ...rawState, attributes: fallbackAttributes };
     }
   }
   // Inject applied only to by-value SOs
-  return attributeService.injectReferences(
+  const newState = attributeService.injectReferences(
     ('attributes' in rawState ? rawState : { attributes: rawState }) as LensRuntimeState,
     references?.length ? references : undefined
   );
+  if (newState.isNewPanel) {
```

Comment:
If it's a new panel, we can safely assume (for now) that is a ES|QL panel.
Therefore try to load the attributes from the query here